domain/application/service/application_test.go

```diff
@@ -91,24 +127,44 @@ func (s *applicationServiceSuite) TestCreateWithStorageBlock(c *gc.C) {
 	pool := domainstorage.StoragePoolDetails{Name: "loop", Provider: "loop"}
 	s.state.EXPECT().GetStoragePoolByName(gomock.Any(), "loop").Return(pool, nil)
 
-	a := AddUnitParams{
+	a := AddUnitArg{
 		UnitName: ptr("foo/666"),
 	}
-	err := s.service.CreateApplication(context.Background(), "666", AddApplicationParams{
-		Charm: s.charm,
-	}, a)
+	_, err := s.service.CreateApplication(context.Background(), "666", s.charm, AddApplicationArgs{}, a)
 	c.Assert(err, jc.ErrorIsNil)
 }
 
 func (s *applicationServiceSuite) TestCreateWithStorageBlockDefaultSource(c *gc.C) {
 	defer s.setupMocks(c).Finish()
 
-	u := application.AddUnitParams{
+	id := applicationtesting.GenApplicationUUID(c)
+
+	u := application.AddUnitArg{
 		UnitName: ptr("foo/666"),
 	}
+	ch := domaincharm.Charm{
+		Metadata: domaincharm.Metadata{
+			Name:  "foo",
+			RunAs: "default",
+			Storage: map[string]domaincharm.Storage{
+				"data": {
+					Name:        "data",
+					Type:        domaincharm.StorageBlock,
+					Shared:      false,
+					CountMin:    1,
+					CountMax:    2,
+					MinimumSize: 10,
+				},
+			},
+		},
+	}
 	s.state.EXPECT().StorageDefaults(gomock.Any()).Return(domainstorage.StorageDefaults{DefaultBlockSource: ptr("fast")}, nil)
-	s.state.EXPECT().UpsertApplication(gomock.Any(), "666", u).Return(nil)
+	s.state.EXPECT().CreateApplication(gomock.Any(), "666", ch, u).Return(id, nil)
```

Comment:
Could just about about add a helper for this block.

Reply:
Something for the future.